nncf/experimental/common/tensor_statistics/collectors.py

```diff
@@ -777,17 +780,60 @@ def _register_reduced_input_impl(self, x: TensorType) -> None:
         return self._container.append(x)
 
     def _aggregate_impl(self) -> Dict[float, NNCFTensor]:
-        stacked_val = self._tensor_processor.stack(self._container)
+        stacked_val, shape_after_aggregation = _moveaxes_flatten_cat(
+            self._container, [x - 1 for x in self._aggregation_axes if x > 0], self._tensor_processor
+        )
 
         percentiles = self._tensor_processor.percentile(
-            stacked_val, self._percentiles_to_collect, axis=self._aggregation_axes, keepdims=self._keepdims
+            stacked_val, self._percentiles_to_collect, axis=0, keepdims=False
         )
         retval = {}
         for idx, percentile in enumerate(self._percentiles_to_collect):
-            retval[percentile] = percentiles[idx].tensor
+            value = percentiles[idx]
+            if self._keepdims:
+                value = self._tensor_processor.reshape(value, shape_after_aggregation)
+            retval[percentile] = value.tensor
         return retval
 
 
+def _moveaxes_flatten_cat(
```

Comment:
This function is working correctly only for cases when 0 is present in original aggregation axes. I didn't implement additional logic for other cases because it will be impossible after the  https://github.com/openvinotoolkit/nncf/pull/2213 anyway